src/Bridge/Doctrine/MongoDbOdm/CollectionDataProvider.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm;
+
+use ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm\Extension\AggregationCollectionExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm\Extension\AggregationResultCollectionExtensionInterface;
+use ApiPlatform\Core\DataProvider\CollectionDataProviderInterface;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use ApiPlatform\Core\Exception\RuntimeException;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ODM\MongoDB\Aggregation\Builder;
+
+/**
+ * Collection data provider for the Doctrine MongoDB ODM.
+ *
+ * @experimental
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class CollectionDataProvider implements CollectionDataProviderInterface, RestrictedDataProviderInterface
+{
+    private $managerRegistry;
+    private $collectionExtensions;
+
+    /**
+     * @param AggregationCollectionExtensionInterface[] $collectionExtensions
+     */
+    public function __construct(ManagerRegistry $managerRegistry, iterable $collectionExtensions = [])
+    {
+        $this->managerRegistry = $managerRegistry;
+        $this->collectionExtensions = $collectionExtensions;
+    }
+
+    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
+    {
+        return null !== $this->managerRegistry->getManagerForClass($resourceClass);
```

Comment:
Since you're dealing with a manager registry from `doctrine/persistence`, you'll want to check whether the manager returned for `$resourceClass` is a document manager.